crates/fluvio-protocol/src/core/bytebuf.rs

```diff
@@ -0,0 +1,301 @@
+use std::io::Error;
+use std::io::ErrorKind;
+
+use bytes::buf::UninitSlice;
+use bytes::{Buf, Bytes, BufMut};
+
+use crate::{Encoder, Decoder, Version};
+
+/// Represnts a SmartModule WASM File bytes.
+///
+/// Provides a `Encoder` and `Decoder` implementation optimized for WASM files
+/// used in SmartModules.
+#[derive(Clone, Debug, Default, PartialEq, Eq)]
+pub struct ByteBuf {
+    inner: Vec<u8>,
+}
+
+impl ByteBuf {
+    pub fn is_empty(&self) -> bool {
```

Comment:
You can use Deref which let you get rid of wrapper methods

Reply:
Addressed on: https://github.com/infinyon/fluvio/pull/2738/commits/6efaa0b83cbfb1f2901dff646b27a688e2422cbd